### PR TITLE
Fix Client Infer Timer. Returning reference of the timer for starting / stopping the timer

### PR DESCRIPTION
**Pull request by GuanLuo**

### CODE REVIEW

File: src/clients/c++/request_common.h [hunk 1]
```diff
@@ -331,7 +331,7 @@ class RequestImpl : public InferContext::Request {
   bool IsReady() const { return ready_; }
   void SetIsReady(bool r) { ready_ = r; }
 
-  RequestTimers Timer() const { return timer_; }
+  RequestTimers& Timer() { return timer_; }
```
Comment (deadeyegoodwin) [inline src/clients/c++/request_common.h]: Why is this change needed all by itself?

Comment (GuanLuo) [reply src/clients/c++/request_common.h]: The timer is used to record time like the following:
`current_context->Timer().Record(RequestTimers::Kind::REQUEST_START);`
The original function call returns a copy of the timer, and thus the timer doesn't get set properly. See #242 description, gRPC time is all zeros.

Comment (deadeyegoodwin) [reply src/clients/c++/request_common.h]: I don't see any description in that other pull request but perhaps I'm looking in the wrong place. We should have some perf_client testing that makes sure that the timers don't just return all zeros. Can you add that to this PR.
